### PR TITLE
fix: 🐛 IOSSDKBUG-430 Fiori FilterFeedbackBar enhancement

### PR DESCRIPTION
**Pull request by xiaoqinggrace**
Feature change request for showing the label when there is a single item selected.
Based on the requirement, change the showsValueForSingleSelected to singleSelectedMode. This will allow the Filter Feedback Bar label to show the name, value, or name and value when a single item is selected.
In the following screenshot, the 2nd, 3rd, and 4th cells are set as show name, show value, and show name and value.
fixedFilterFeedbackBar

### CODE REVIEW

File: Sources/FioriSwiftUICore/DataTypes/SortFilter+DataType.swift [hunk 1]
```diff
@@ -309,7 +309,7 @@ public extension SortFilterItem {
         var valueOptions: [String]
         public let allowsMultipleSelection: Bool
         public let allowsEmptySelection: Bool
-        public var showsValueForSingleSelected: Bool = true
```
Comment (billzhou0223) [inline Sources/FioriSwiftUICore/DataTypes/SortFilter+DataType.swift:312]: Use @available keyword to deprecate a public attribute.

File: Sources/FioriSwiftUICore/DataTypes/SortFilter+DataType.swift [hunk 2]
```diff
@@ -327,7 +327,17 @@ public extension SortFilterItem {
             case list
         }
         
-        public init(id: String = UUID().uuidString, name: String, value: [Int], valueOptions: [String], allowsMultipleSelection: Bool, allowsEmptySelection: Bool, showsValueForSingleSelected: Bool = true, icon: String? = nil, itemLayout: OptionListPickerItemLayoutType = .fixed, displayMode: DisplayMode = .automatic) {
+        /// Enum for display modes of the filter feed bar label when a single item selected
+        public enum SingleSelectedMode {
```
Comment (billzhou0223) [inline Sources/FioriSwiftUICore/DataTypes/SortFilter+DataType.swift:331]: Suggest calling it `barItemDisplayMode`

File: Sources/FioriSwiftUICore/DataTypes/SortFilter+DataType.swift [hunk 3]
```diff
@@ -327,7 +327,17 @@ public extension SortFilterItem {
...
+            /// To show the name
+            case showName
+            /// To show the value
+            case showValue
+            /// To show both the name and value
+            case showNameValue
```
Comment (billzhou0223) [inline Sources/FioriSwiftUICore/DataTypes/SortFilter+DataType.swift:337]: Change to 

```
            case name
            /// To show the value
            case value
            /// To show both the name and value
            case nameAndValue
```

**Review by billzhou0223 — CHANGES_REQUESTED** [review]

File: Sources/FioriSwiftUICore/DataTypes/SortFilter+DataType.swift [hunk 4]
```diff
@@ -327,7 +327,17 @@ public extension SortFilterItem {
             case list
         }
         
-        public init(id: String = UUID().uuidString, name: String, value: [Int], valueOptions: [String], allowsMultipleSelection: Bool, allowsEmptySelection: Bool, showsValueForSingleSelected: Bool = true, icon: String? = nil, itemLayout: OptionListPickerItemLayoutType = .fixed, displayMode: DisplayMode = .automatic) {
+        /// Enum for display modes of the filter feed bar label when a single item selected
```
Comment (dyongxu) [inline Sources/FioriSwiftUICore/DataTypes/SortFilter+DataType.swift:330]:
```suggestion
        /// Enum for display mode of the FilterFeedbackBar item when only one value is selected for that item
        /// This is effective, regardless of that item allowing multiple selection or not
```

File: Sources/FioriSwiftUICore/DataTypes/SortFilter+DataType.swift (hunk 2 above)
Comment (dyongxu) [inline Sources/FioriSwiftUICore/DataTypes/SortFilter+DataType.swift:331]:
```suggestion
        public enum SingleValueDisplayMode {
```

File: Sources/FioriSwiftUICore/DataTypes/SortFilter+DataType.swift (hunk 3 above)
Comment (dyongxu) [inline Sources/FioriSwiftUICore/DataTypes/SortFilter+DataType.swift:337]:
```suggestion
            /// To show the name only
            case showNameOnly
            /// To show the value only
            case showValueOnly
            /// To show both the name and value
            case showNameAndValue
```

**Review by dyongxu — CHANGES_REQUESTED** [review]
1. Comments inlined
2. May need to add some doc what is the default behavior regarding single value selection mode. I assume the default is "NameAndValue", sth like "To-Do(1)

File: Sources/FioriSwiftUICore/DataTypes/SortFilter+DataType.swift [hunk 5]
```diff
@@ -418,14 +428,21 @@ public extension SortFilterItem {
...
+            if self.value.count == 1 {
+                switch self.singleSelectedMode {
+                case .showName:
+                    return self.name
+                case .showValue:
                     return self.valueOptions[self.value[0]]
-                } else {
-                    return "\(self.name) (\(self.value.count))"
+                case .showNameValue:
+                    return self.name + ":" + self.valueOptions[self.value[0]]
```
Comment (dyongxu) [inline Sources/FioriSwiftUICore/DataTypes/SortFilter+DataType.swift:438]:
```suggestion
                    return self.name + ": " + self.valueOptions[self.value[0]]
```

File: Sources/FioriSwiftUICore/DataTypes/SortFilter+DataType.swift (hunk 1 above)
Comment (xiaoqinggrace) [reply Sources/FioriSwiftUICore/DataTypes/SortFilter+DataType.swift:312]: Based on the feature request, showsValueForSingleSelected was added three weeks ago. There is no new build for rel-4.1 after it was added.

File: Sources/FioriSwiftUICore/DataTypes/SortFilter+DataType.swift (hunk 2 above)
Comment (xiaoqinggrace) [reply Sources/FioriSwiftUICore/DataTypes/SortFilter+DataType.swift]: Bill also suggested the name. Which one should I follow?

File: Sources/FioriSwiftUICore/DataTypes/SortFilter+DataType.swift (hunk 3 above)
Comment (xiaoqinggrace) [reply Sources/FioriSwiftUICore/DataTypes/SortFilter+DataType.swift]: Bill also suggested the change. Which one should I follow?

**Review by dyongxu — APPROVED** [review]

**Review by billzhou0223 — APPROVED** [review]

**Pull request merged**